### PR TITLE
Discord implementation

### PR DESCRIPTION
**Pull request by Time6628**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -32,15 +32,35 @@ compileJava {
...
+    relocate 'com.neovisionaries.ws.client', 'io.github.nucleuspowered.phonon.relocate.com.neovisionaries.ws.client'
+    relocate 'net.dv8tion.jda', 'io.github.nucleuspowered.phonon.relocate.net.dv8tion.jda'
+
     archiveName = "Phonon-${version}-plugin.jar"
 }
 
 dependencies {
-    compile("org.spongepowered:spongeapi:5.1.0")
-    compile("net.dv8tion:JDA:3.0.BETA2_108")
+    compile "org.spongepowered:spongeapi:5.1.0"
```
Comment (Mohron) [inline build.gradle]: Should be consistent with your quotes. All other lines use single quotes.

File: src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import net.dv8tion.jda.core.entities.*;
+import net.dv8tion.jda.core.exceptions.RateLimitedException;
+
+import javax.security.auth.login.LoginException;
+import java.awt.*;
+import java.util.List;
+import java.util.Optional;
+
+public class DiscordBot {
+    static DiscordBot instance;
```
Comment (Mohron) [inline src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java]: This has a getter and can/should be private.

File: src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    JDA jda;
```
Comment (Mohron) [inline src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java]: This can also be private.

File: build.gradle (hunk 1 above)
Comment (Time6628) [inline build.gradle]: Sorry, will fix.

File: src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+import javax.security.auth.login.LoginException;
+import java.awt.*;
+import java.util.List;
+import java.util.Optional;
+
+public class DiscordBot {
+    private static DiscordBot instance;
+
+    private JDA jda;
```
Comment (dualspiral) [inline src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java]: If a user was to change their token and reload, would we be OK with just replacing this instance? If so, `jda` should be final, I believe there is a _slight_ performance boost to making it final.

Comment (Time6628) [inline src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java]: This is not for the users, this is the bot that will handle the users.

Comment (dualspiral) [inline src/main/java/io/github/nucleuspowered/phonon/discord/DiscordBot.java]: Sorry, I meant server owner!

**Comment by dualspiral** [comment]
Alright, I'll squash and merge.

**Comment by dualspiral** [comment]
https://github.com/NucleusPowered/Phonon/commit/deb86edd8bb47127abdeabe8e0fe3339e9a6e7b4

**Pull request closed without merging**